server/src/main/java/org/opensearch/common/regex/Regex.java

```diff
@@ -125,39 +126,7 @@ public static boolean simpleMatch(String pattern, String str, boolean caseInsens
             pattern = Strings.toLowercaseAscii(pattern);
             str = Strings.toLowercaseAscii(str);
         }
-        return simpleMatchWithNormalizedStrings(pattern, str);
-    }
-
-    private static boolean simpleMatchWithNormalizedStrings(String pattern, String str) {
-        int sIdx = 0, pIdx = 0, match = 0, wildcardIdx = -1;
-        while (sIdx < str.length()) {
-            // both chars matching, incrementing both pointers
-            if (pIdx < pattern.length() && str.charAt(sIdx) == pattern.charAt(pIdx)) {
-                sIdx++;
-                pIdx++;
-            } else if (pIdx < pattern.length() && pattern.charAt(pIdx) == '*') {
-                // wildcard found, only incrementing pattern pointer
-                wildcardIdx = pIdx;
-                match = sIdx;
-                pIdx++;
-            } else if (wildcardIdx != -1) {
-                // last pattern pointer was a wildcard, incrementing string pointer
-                pIdx = wildcardIdx + 1;
-                match++;
-                sIdx = match;
-            } else {
-                // current pattern pointer is not a wildcard, last pattern pointer was also not a wildcard
-                // characters do not match
-                return false;
-            }
-        }
-
-        // check for remaining characters in pattern
-        while (pIdx < pattern.length() && pattern.charAt(pIdx) == '*') {
-            pIdx++;
-        }
-
-        return pIdx == pattern.length();
+        return Glob.globMatch(pattern, str);
```

Comment:
Thank you for this change. I think this is much more reflective of what this method actually does. Regex is a misnomer for this class.

Reply:
I agree!

Reply:
> Regex is a misnomer for this class.

+1, I was wondering about the same thing.